packages/at_chops/lib/src/algorithm/algo_type.dart

```diff
@@ -1,4 +1,26 @@
 // ignore: constant_identifier_names
+import 'package:at_commons/at_commons.dart';
+
 enum SigningAlgoType { ecc_secp256r1, rsa2048, rsa4096 }
 
-enum HashingAlgoType { sha256, sha512, md5 }
+enum HashingAlgoType {
+  sha256,
+  sha512,
+  md5,
+  argon2id;
+
+  static HashingAlgoType fromString(String name) {
+    switch (name.toLowerCase()) {
```

Comment:
you can replace impl with
return HashingAlgoType.values.firstWhere(
      (algo) => algo.name == value,
      orElse: () => throw Exception,  
    );